cloudinit/sources/DataSourceWSL.py

```diff
@@ -237,23 +268,84 @@ def check_instance_id(self, sys_cfg) -> bool:
             return False
 
     def _get_data(self) -> bool:
-        self.vendordata_raw = None
-        seed_dir = cloud_init_data_dir()
+        if not subp.which(WSLPATH_CMD):
+            LOG.debug(
+                "No WSL command %s found. Cannot detect WSL datasource",
+                WSLPATH_CMD,
+            )
+            return False
+        self.instance_name = instance_name()
+
+        try:
+            user_home = find_home()
+        except IOError as e:
+            LOG.debug("Unable to detect WSL datasource: %s", e)
+            return False
 
+        seed_dir = cloud_init_data_dir(user_home)
+        user_data: Optional[Union[dict, bytes]] = None
+
+        # Load any metadata
         try:
             self.metadata = load_instance_metadata(
                 seed_dir, self.instance_name
             )
-            file = self.find_user_data_file(seed_dir)
-            self.userdata_raw = cast(
-                str, util.load_binary_file(file.as_posix())
-            )
-            return True
+        except (ValueError, IOError) as err:
+            LOG.error("Unable to load metadata: %s", str(err))
+            return False
+
+        # Load Ubuntu Pro configs
+        agent_data, user_data = load_ubuntu_pro_data(user_home)
```

Comment:
I just realized we don't check if the distro is Ubuntu before doing this. Sorry for not raising that before, but we should just skip this step if the distro is not Ubuntu, as it contains config data that's not relevant to other distros, such as an Ubuntu Pro subscription token.